### PR TITLE
added registerChangeNotifierHandler and fixed some typo. closes #19

### LINKED ISSUE #19: Cannot register handler for ChangeNotifiers
I have a _Counter_ class:

```
class Counter extends ChangeNotifier {
        int _count = 0;
        int get count=>_count;

        void increment(){
              _count++;
             notifyListeners();
       }
}
```

Now I want to show a dialog when the value of the `count` becomes greater than 5. 
Therefore I want to register a handler in the build method like:

```
  Widget build(BuildContext context) {
    registerHandler(
        select: (Counter c) => c.count,
        handler: (context, value, cancel) => showNameDialog(context, value));
    ...
  }
```

But it says _count is not a ValueListenable as required by the context_ or something like that. I can watch it though.

It is cumbersome to wrap every variable in a `ValueListenable`. And because we can watch it we should also be able to register handlers for it.

### PR DESCRIPTION
**Pull request by smjxpro**

### CODE REVIEW

File: lib/src/watch_it.dart [hunk 1]
```diff
@@ -268,11 +268,11 @@ AsyncSnapshot<R?> watchFuture<T extends Object, R>(
 
 /// [registerHandler] registers a [handler] function for a `ValueListenable`
 /// exactly once on the first build
-/// and unregisters it when the widget is destroyed.
```
Comment (escamoteur) [inline lib/src/watch_it.dart:271]: I think registers is correct here as the acting 'person' is watch_it which would be 3rd person

**Review by escamoteur — APPROVED** [review]
Awesome work, will merge as it is and change some of the corrected verbs where you removed the trailing 's'

**Comment by escamoteur** [comment]
new version is up on pub

**Comment by smjxpro** [comment]
@escamoteur I just read the changelog on pub.dev which says "you can now register handlers for pure Listenables/ChangeNotifiers", isn't it inconrrect because I constrained the object to be a ChangeNotifier because I didn't have listenables in my mind. However, after reading the log, the idea came to my head that I could just constrain it to be a Listenable and change the handler name to be registerListenableHandler for all the listenables including ChangeNotifier.
Can you do that? Or should I do it and create another PR?

**Comment by escamoteur** [comment]
If you have time it would be great if you could make another PR
Am 27. Dez. 2023, 14:21 +0100 schrieb S. M. JAHANGIR ***@***.***>:

**Comment by smjxpro** [comment]
> If you have time it would be great if you could make another PR
> Am 27. Dez. 2023, 14:21 +0100 schrieb S. M. JAHANGIR ***@***.***>:
> > @escamoteur I just read the changelog on pub.dev which says "you can now register handlers for pure Listenables/ChangeNotifiers", isn't it inconrrect because I constrained the object to be a ChangeNotifier because I didn't have listenables in my mind. However, after reading the log, the idea came to my head that I could just constrain it to be a Listenable and change the handler name to be registerListenableHandler for all the listenables including ChangeNotifier.
> > Can you do that? Or should I do it and create another PR?
> > —
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> > You are receiving this because you were mentioned.Message ID: ***@***.***>
> 

Ok. I am doing it. 😀

**Comment by smjxpro** [comment]
On another note, let's say I am throwing an exception inside one of the methods in my ChangeNotifier, is there any way or is it a good idea to catch that exception and pass the error on another callback in watchers/handlers?

**Comment by escamoteur** [comment]
IMHO if you expect an exception you should place the try catch inside your handler function
Am 27. Dez. 2023, 14:26 +0100 schrieb S. M. JAHANGIR ***@***.***>:

**Comment by smjxpro** [comment]
@escamoteur After searching I found out that only `ValueNotifier` and `ChangeNotifier` implements the `Listenable` interface which are used for state management purposes. And there's already a handler for `ValueNotifier` therefore I found it redundant to implement/generalize another on by changing `registerChangeNotifierHnadler` to `registerListenableHandler`.

It just needs to change the **you now can register handlers for pure Listenable/ChangeNotifiers too** to **you now can register handlers for ChangeNotifiers too** in the changelog.

Also, it would be useful to add the pub.dev link in the about section on the repo.

Thanks for the great work.

**Comment by escamoteur** [comment]
It depends, anybody can implement their own listenable class, but you are probably right, let's keep it like that for the moment
Am 27. Dez. 2023, 15:38 +0100 schrieb S. M. JAHANGIR ***@***.***>:

**Pull request merged**